features/ui/button/button.tsx

```diff
@@ -1,9 +1,45 @@
-import { ButtonHTMLAttributes } from "react";
+import React, { ButtonHTMLAttributes, ReactElement } from "react";
 import classNames from "classnames";
 import styles from "./button.module.scss";
 
-export function Button(props: ButtonHTMLAttributes<HTMLButtonElement>) {
+interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
+  size?: "small" | "medium" | "large" | "xlarge";
+  color?:
+    | "primary"
+    | "secondary"
+    | "gray"
+    | "empty"
+    | "empty-gray"
+    | "error"
+    | "empty-error";
+  icon?: ReactElement;
+  iconPosition?: "leading" | "trailing";
+}
+
+export function Button({
+  size,
+  color,
+  icon,
+  iconPosition = "leading",
+  children,
+  ...props
+}: ButtonProps) {
+  const hasIcon = icon ? "hasIcon" : "";
+
   return (
-    <button {...props} className={classNames(styles.button, props.className)} />
+    <button
+      {...props}
+      className={classNames(
+        styles.button,
+        props.className,
+        styles[size || ""],
+        styles[color || ""],
```

Comment:
I'd probably set default values for these props instead of the empty string. Or what happens if one of those isn't set? Is that the unstyled button then? That would be an elegant solution actually